src/snapred/backend/data/Indexer.py

```diff
@@ -186,65 +183,33 @@ def nextVersion(self) -> int:
-        if set(self.index.keys()) == self.dirVersions:
-            # remove the default version
-            dirVersions = [x for x in self.dirVersions if x != VERSION_DEFAULT]
-            # if nothing is left, the next is the start
-            if len(dirVersions) == 0:
-                version = VERSION_START
-            # otherwise, the next is max version + 1
-            else:
-                version = max(dirVersions) + 1
-        # if the index and directory are out of sync, find the largest in both sets
-        else:
-            # get the elements particular to each set -- the max of these is the next version
-            indexSet = set(self.index.keys())
-            diffAB = indexSet.difference(self.dirVersions)
-            diffBA = self.dirVersions.difference(indexSet)
-            # if diffAB is nullset, diffBA has one more member -- that is next
-            if diffAB == set():
-                version = list(diffBA)[0]
-            # if diffBA is nullset, diffAB has one more member -- that is next
-            elif diffBA == set():
-                version = list(diffAB)[0]
-            # otherwise find the max of both differences and return that
-            else:
-                indexVersion = max(diffAB)
-                directoryVersion = max(diffBA)
-                version = max(indexVersion, directoryVersion)
-
-        return version
-
-    @validate_call
-    def thisOrCurrentVersion(self, version: Optional[int]):
-        if self.isValidVersion(version):
-            return version
-        else:
-            return self.currentVersion()
+        if set(self.index.keys()) != self.dirVersions:
+            self.reconcileIndexToFiles()
 
-    @validate_call
-    def thisOrNextVersion(self, version: Optional[int]):
-        if self.isValidVersion(version):
-            return version
+        if self.currentVersion() is None:
+            return self.defaultVersion()
         else:
-            return self.nextVersion()
+            return self.currentVersion() + 1
 
     @validate_call
-    def thisOrLatestApplicableVersion(self, runNumber: str, version: Optional[int]):
-        if self.isValidVersion(version) and self._isApplicableEntry(self.index[version], runNumber):
-            return version
-        else:
-            return self.latestApplicableVersion(runNumber)
+    def getLatestApplicableVersion(self, runNumber: str):
+        return self.latestApplicableVersion(runNumber)
 
-    def isValidVersion(self, version):
+    def validateVersion(self, version):
         try:
-            VersionedObject.parseVersion(version, exclude_none=True)
+            VersionedObject(version=version)
             return True
         except ValueError:
-            return False
+            # This error would only ever result from a software bug.
+            # Saving/Loading/Refering to erroneous "current" versions just serves to obfuscate the error.
```

Comment:
I'm pretty sure it could also result from someone manually editing either the index or record files

Reply:
but then we wouldnt be able to read said files as they would fail to serialize into a `VersionedObject`, thus would not reach this portion of the code. (fail to read probably means we needed to migrate versions or something)

Since it is validated on write/read, this step only applies to externally supplied input, which means either:
1. We failed to validate user input and it somehow made it this far down without error
2. We wrote code that submitted an invalid version(Bad coversion, typo, Point 1, ect)